### PR TITLE
Support Large Saturation Function Tables in INIT File

### PR DESCRIPTION
**Pull request by bska**
This PR relaxes the INIT file writing code's assumption that tabulated saturation functions have at most `NSSFUN` (`TABDIMS(3)`, [default value 20](https://github.com/OPM/opm-common/blob/138e355a0a96c6a8a0af3d2f6d13e9addc5744fd/opm/input/eclipse/share/keywords/000_Eclipse100/T/TABDIMS#L20-L25)) rows per saturation region.  The rest of the simulator code does not assume this restriction and there's no particular reason why the `TABDIMS`/`TAB` array should impose this either.  Moreover, we already relax similar assumptions for the PVT tables.  Following this PR `NSSFUN` becomes a "minimum maximum" number of rows in the TAB array output of the run's saturation functions.

To this end, switch to returning a three-element `tuple<>` of the form
```C++
tuple<int, int, vector<double>>
```
from each function which normalises individual saturation function keywords.  We then treat this return type as follows

  * Element at index 0 (`get<0>(t)`) is the number of allocated rows in each individual table.  This is usually `NSSFUN`, but we allow input tables larger than this value.  In that case, the number of allocated rows will correspond to the maximum number of active table rows in the keyword's input data across all saturation regions.

  * Element at index 1 (`get<1>(t)`) is the number of individual tables&ndash;i.e., the number of saturation regions.  This is mostly redundant information but provided as a simplification for client code.

  - Element at index 2 (`get<2>(t)`) is normalised and linearised table data for the corresponding saturation functions.

We additionally create a number of intermediate helper functions named `normalise*()` which inspect the run's saturation function family and active phases and calls the appropriate keyword functions depending on which normalised table we're forming.  This simplifies the implementation of `Tables::addSatFunc()`.

### CODE REVIEW

**Comment by bska** [comment]
I am creating this PR in draft mode because I want to have wider testing before we enable this in the mainline.

**Comment by bska** [comment]
jenkins build this please

File: opm/output/eclipse/Tables.cpp [hunk 1]
```diff
@@ -364,147 +489,246 @@ namespace { namespace SatFunc {
...
+                                letTab[(1+dOff)*offset + k*numRows + i - 1]) / ds;
+                       }
+                   }
+               }
+           }
+
+            return letTab;
+        }
+
+        /// Infer number of alloated rows in normalised saturation function
```
[the remark this answers is not in the document]
Comment (bska) [reply opm/output/eclipse/Tables.cpp]: > allocated

Thanks.  I've pushed an update to address this.

**Comment by bska** [comment]
This work has been through considerable local testing and I'm now sufficiently satisfied that it does what it's supposed to do.  I'm marking the PR as "ready for review" and I'm running a build check.

**Comment by bska** [comment]
jenkins build this

**Review by akva2 — APPROVED** [review]
It's a rather large change, but I have read it a few times now and nothing pops out. You sound confident in the testing done so let's get it in.

**Comment by akva2** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
PR approved and build check is green.  I'll merge into master.

**Pull request merged**